s3tests_boto3/functional/test_s3select.py

```diff
@@ -0,0 +1,410 @@
+                row = row + "{}{:02d}{:02d}-{:02d}{:02d}{:02d},".format(random.randint(0,100)+1900,random.randint(1,12),random.randint(1,28),random.randint(0,23),random.randint(0,59),random.randint(0,59),);
+            result += row + "\n"
+
+        return result
+
+def create_random_csv_object(rows,columns,col_delim=",",record_delim="\n",csv_schema=""):
+        result = ""
+        if len(csv_schema)>0 :
+            result = csv_schema + record_delim
+
+        for i in range(rows):
+            row = "";
+            for y in range(columns):
+                row = row + "{}{}".format(random.randint(0,1000),col_delim);
+            result += row + record_delim
+
+        return result
+
+
+def upload_csv_object(bucket_name,new_key,obj):
+        conn = get_connection()
+        conn.create_bucket( bucket_name )
+        bucket = conn.get_bucket( bucket_name )
+
+        k1 = bucket.new_key( new_key );
+        k1.set_contents_from_string( obj );
+        
+    
+def run_s3select(bucket,key,query,column_delim=",",row_delim="\n",quot_char='"',esc_char='\\',csv_header_info="NONE"):
+
+    s3 = boto3.client('s3',#'sns',
+        endpoint_url=get_config_endpoint(),
+        aws_access_key_id=get_main_aws_access_key(),
+        region_name=region_name,
+        aws_secret_access_key=get_main_aws_secret_key())
+        #config=Config(signature_version='v2'))
+
+
+    r = s3.select_object_content(
+        Bucket=bucket,
+        Key=key,
+        ExpressionType='SQL',
+        InputSerialization = {"CSV": {"RecordDelimiter" : row_delim, "FieldDelimiter" : column_delim,"QuoteEscapeCharacter": esc_char, "QuoteCharacter": quot_char, "FileHeaderInfo": csv_header_info}, "CompressionType": "NONE"},
+        OutputSerialization = {"CSV": {}},
+        Expression=query,)
+    
+    result = ""
+    for event in r['Payload']:
+        if 'Records' in event:
+            records = event['Records']['Payload'].decode('utf-8')
+            result += records
+    
+    return result
+
+def remove_xml_tags_from_result(obj):
+    result = ""
+    for rec in obj.split("\n"):
+        if(rec.find("Payload")>0 or rec.find("Records")>0):
+            continue
+        result += rec + "\n" # remove by split
+
+    return result
+
+def create_list_of_int(column_pos,obj):
+    res = 0
+    list_of_int = [] 
+    for rec in obj.split("\n"):
+        col_num = 1
+        if ( len(rec) == 0):
+            continue;
+        for col in rec.split(","):
+            if (col_num == column_pos):
+                list_of_int.append(int(col));
+            col_num+=1; 
+
+    return list_of_int
+       
+def test_count_operation():
+    csv_obj_name = "csv_star_oper"
+    bucket_name = "test"
+    num_of_rows = 10
+    obj_to_load = create_random_csv_object(num_of_rows,10)
+    upload_csv_object(bucket_name,csv_obj_name,obj_to_load)
+    res = remove_xml_tags_from_result( run_s3select(bucket_name,csv_obj_name,"select count(0) from stdin;") ).replace(",","")
```

Comment:
is it expected by the spec to have the extra comma at the end of the result set?

Reply:
output serialization is not implemented (only input serialization is implemented, part of it).
the comma at the end, is not part of spec (row delimiter , field delimiter should be parameterized)

Reply:
if the comma at the end is not expected, can we remove it?

Reply:
ok